java/src/test/java/org/hyperledger/fabric/client/FileCheckpointerTest.java

```diff
@@ -0,0 +1,55 @@
+package org.hyperledger.fabric.client;
+
+
+
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class FileCheckpointerTest extends CommonCheckpointerTest {
+    FileCheckpointer fileCheckpointer;
+    private static final TestUtils testUtils = TestUtils.getInstance();
+
+    @Override
+    protected FileCheckpointer getCheckpointerInstance() throws IOException {
+        Path file = testUtils.createTempFile(".json");
+        this.fileCheckpointer = new FileCheckpointer(file);
+        return fileCheckpointer;
+    }
+
+    @Override
+    protected void tearDown() throws IOException {
+        this.fileCheckpointer.close();
+    }
+
+    @Test
+    void state_is_persisted() throws IOException {
+        Path file = testUtils.createTempFile(".json");
+        this.fileCheckpointer = new FileCheckpointer(file);
+        this.fileCheckpointer.checkpointTransaction(1, Optional.ofNullable("TRANSACTION_ID"));
+        this.fileCheckpointer.close();
```

Comment:
Let's give try-with-resources a try since that's why we implemented AutoCloseable!
```suggestion
        try (fileCheckpointer = new FileCheckpointer(file)) {
            fileCheckpointer.checkpointTransaction(1, Optional.ofNullable("TRANSACTION_ID"));
        }
```